### PR TITLE
Control Registry

### PR DESCRIPTION
**Pull request by duzos**
Converts the `ConsoleControl` enum into a Registry for `Control`s
Allows for other mods to add their own controls.

### CODE REVIEW

**Review by Jeryn99 — COMMENTED** [review]
I am leaving @50ap5ud5 to review this as it is registry related but I am not a fan of abusing the super like this ``super(new ResourceLocation(TardisRefined.MODID, "door_toggle"));``

If a mod wanted to extend one of the existing controls, this would complicate things

**Comment by duzos** [comment]
i love abusing super :(
i will change so theres multiple constructors or smth if wanted then

**Comment by Jeryn99** [comment]
> i love abusing super :( i will change so theres multiple constructors or smth if wanted then

Yea sounds good

File: common/src/main/java/whocraft/tardis_refined/common/tardis/control/Control.java [hunk 1]
```diff
@@ -65,4 +77,27 @@ public boolean canUseControl(TardisLevelOperator tardisLevelOperator, Control co
...
     }
 
+    public ResourceLocation getId(){
+        return this.id;
+    }
+    public String getTranslationKey() {return this.langId;}
+    public String getSerializedName() {
+        // todo Temporary - is bad :(
+        return this.getId().getPath();
+    }
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/control/Control.java]: getSerialializedName looks to be a holdover from when the Control was an Enum, if this is not being used we don't need this.

**Review by 50ap5ud5 — CHANGES_REQUESTED** [review]
Looks great, some minor cleanup and this will be good to go.

**Review by 50ap5ud5 — APPROVED** [review]
LGTM

**Pull request merged**